### PR TITLE
Rename IContextAccessor -> IScopeInstance

### PR DESCRIPTION
**Pull request by HaoK**
cc @yishaigalatzer @divega @davidfowl @lodejard

### CODE REVIEW

File: src/Microsoft.Framework.DependencyInjection/project.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
             "dependencies": {
                 "System.Collections.Concurrent": "4.0.10-beta-*",
                 "System.Linq.Expressions": "4.0-beta-*",
-                "System.Threading": "4.0.10-beta-*",
+                "System.Threading": "4.0.0-beta-*",
```
Comment (davidfowl) [inline src/Microsoft.Framework.DependencyInjection/project.json]: ?

**Comment by davidfowl** [comment]
The only push back I have on this is https://github.com/aspnet/DependencyInjection/issues/158. We should discuss the naming of this new thing in the same context.

File: src/Microsoft.Framework.DependencyInjection/project.json (hunk 1 above)
Comment (HaoK) [inline src/Microsoft.Framework.DependencyInjection/project.json]: Oops bad merge I will fix

**Comment by HaoK** [comment]
Does supporting #158 require any changes to the interface?  Or do we just need to make sure that the name convey's that it can also be used for that scenario?

**Comment by davidfowl** [comment]
@HaoK no changes are required to the interface itself. It might even be another interface name. My only concern was calling both things IScopedSomething<T> might be confusing.

**Comment by HaoK** [comment]
Got it, yeah having IScopeLocal and IScopeData which do different things would be pretty confusing.

Could we rename this one to be IRequestLocal then?  The scope is implicitly the request today isn't it?

**Comment by davidfowl** [comment]
Maybe we just rename the other one?

**Comment by HaoK** [comment]
Yeah the other one isn't specifically related to scope, that one could be ICallContext maybe?  Since its abstracting async local/call context?

**Comment by davidfowl** [comment]
Hmm

**Comment by halter73** [comment]
Would `IScopeValue` be a better name? That way it might be less likely to get confused with whatever we end up doing to abstract the call context.

**Comment by davidfowl** [comment]
:shipit:

**Comment by davidfowl** [comment]
Value seems better tbh

**Comment by HaoK** [comment]
Maybe IScopeInstance.Value?   IScopeValue is ok too

**Comment by davidfowl** [comment]
```C#
IScopedInstance<T> 
```

Sounds better

**Comment by davidfowl** [comment]
:shipit:

**Comment by yishaigalatzer** [comment]
IScopeValue sounds as good as it is going to get :)

**Comment by HaoK** [comment]
I think we settled on IScopedInstance

On Jan 10, 2015, at 4:29 PM, "Yishai Galatzer" <notifications@github.com<mailto:notifications@github.com>> wrote:


IScopeValue sounds as good as it is going to get :)

-

**Comment by yishaigalatzer** [comment]
That's fine too

**Comment by HaoK** [comment]
Merged b3f137eaab3366f3e03b047d1ff6fb40bb26cbdc

**Pull request closed without merging**